src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaderValueCollection.cs

```diff
@@ -34,7 +34,7 @@ namespace System.Net.Http.Headers
     //   - Property TransferEncodingChunked: is set to "true".
     public sealed class HttpHeaderValueCollection<T> : ICollection<T> where T : class
     {
-        private string _headerName;
+        private HeaderDescriptor _descriptor;
```

Comment:
How often are these HttpHeaderValueCollections created?  Looks like this will increase its size by a reference field.